$MirOS: ports/lang/pawn/toolkit/patches/patch-source_amx_amxcons_c,v 1.1.1.1 2008/08/08 15:53:57 tg Exp $
--- source/amx/amxcons.c.orig	Thu Mar  6 13:17:51 2008
+++ source/amx/amxcons.c	Fri Aug  8 19:19:11 2008
@@ -90,6 +90,8 @@
   #define EOL_CHAR     '\n'
 #endif
 
+static void acceptchar(int, int *);
+
 #if !defined AMX_STRING_LIB
 
 #if defined AMX_TERMINAL
@@ -1135,13 +1137,8 @@ static cell AMX_NATIVE_CALL n_getstring(
 
     c=amx_getch();
     while (c!=EOF && c!=EOL_CHAR && chars<max-1) {
-      str[chars++]=(TCHAR)c;
-      #if defined(SUPPRESS_ECHO)
- 	    /* For Mac OS X, non-Curses, don't echo the character */
-      #else
-        amx_putchar((TCHAR)c);
-        amx_fflush();
-      #endif
+      str[chars]=(TCHAR)c;
+      acceptchar(c, &chars);
       if (chars<max-1)
         c=amx_getch();
     } /* while */
@@ -1162,9 +1159,11 @@ static void acceptchar(int c,int *num)
 {
   switch (c) {
   case '\b':
+  case '\177':
     amx_putchar('\b');
-    *num-=1;
-    #if defined amx_putchar && (defined __BORLANDC__ || defined __WATCOMC__)
+    if ((*num-=1) < 0)
+      *num = 0;
+    #if defined amx_putchar && (defined __BORLANDC__ || defined __WATCOMC__ || defined VT100)
       /* the backspace key does not erase the
        * character, so do this explicitly */
       amx_putchar(' ');     /* erase */
@@ -1234,7 +1233,7 @@ static cell AMX_NATIVE_CALL n_getvalue(A
     } /* if */
 
     /* check end of input */
-    #if defined __LINUX__ || defined __FreeBSD__ || defined __OpenBSD__ || defined MACOS
+    #if defined __LINUX__ || defined BSD || defined MACOS
       if (c=='\n' && inlist(amx,'\r',params+2,(int)params[0]/sizeof(cell)-1)!=0)
         c='\r';
     #endif
